bluejay-parser/src/ast/variable.rs

```diff
@@ -8,6 +8,12 @@ pub struct Variable<'a> {
     span: Span,
 }
 
+impl<'a> Variable<'a> {
+    pub fn new(name: VariableName<'a>, span: Span) -> Self {
```

Comment:
Probably doesn't need to be fully public?
```suggestion
    pub(crate) fn new(name: VariableName<'a>, span: Span) -> Self {
```